### PR TITLE
feat: add audit logging for rollout actions

### PR DESCRIPTION
**Pull request by yquansah**
Add audit logging for `rollout` type, for the `create`, `update`, and `delete` action.

Completes FLI-457

### CODE REVIEW

File: internal/server/audit/types.go [hunk 1]
```diff
@@ -150,3 +150,46 @@ func NewRule(r *flipt.Rule) *Rule {
...
+	Segment      *RolloutSegment   `json:"segment,omitempty"`
+}
+
+type RolloutThreshold struct {
+	Percentage float32 `json:"percentage"`
+	Value      bool    `json:"value"`
+}
+
+type RolloutSegment struct {
+	Key   string `json:"Key"`
```
Comment (markphelps) [inline internal/server/audit/types.go:169]:
```suggestion
	Key   string `json:"key"`
```

File: internal/server/middleware/grpc/middleware_test.go [hunk 2]
```diff
@@ -1422,6 +1422,97 @@ func TestAuditUnaryInterceptor_DeleteConstraint(t *testing.T) {
...
+	ctx, span := tr.Start(context.Background(), "OnStart")
+
+	got, err := unaryInterceptor(ctx, req, info, handler)
+	require.NoError(t, err)
+	assert.NotNil(t, got)
+
+	span.End()
+	assert.Equal(t, 1, exporterSpy.GetSendAuditsCalled())
+}
+
```
Comment (markphelps) [inline internal/server/middleware/grpc/middleware_test.go:1515]: mind adding a test for delete?

Comment (yquansah) [reply internal/server/middleware/grpc/middleware_test.go:1515]: @markphelps Right! will do that

**Review by markphelps — APPROVED** [review]
nice!

**Pull request merged**